build.rs

```diff
@@ -3,6 +3,11 @@ use std::io::Write;
 use std::{env, fs, path::PathBuf};
 
 // Keep in sync with Cargo.toml.
+//
+// We don't populate this automatically from the Cargo.toml at build time
```

Comment:
Shower thought:

If the rust crate's build.rs did `println!("cargo:version_number={}", env!("CARGO_PKG_VERSION"));` then we could simply use `env!("DEP_RUSTLS_VERSION_NUMBER")` here.

Haven't tried that, and it's probably a worse overall solution than what we have in this PR, and also I should have learned my lesson by now rather than relying on underused parts of cargo.

Reply:
Ooo, I forgot that `build.rs` can forward information like that. Seems worth playing with but we'd need a rustls release with that magic in-place to be usable here so I think I'll leave it for a "One Day" follow-up.